File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Games.java
package com.logistics.Components.Product;

public class Games extends Product {

    protected String genre;

    public Games() {
        super();
        genre = "";
    }

    public Games(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String genre) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit);
        this.genre = genre;
    }

    // Copy Constructor
    public Games(Games other) {
        super(other);
        this.genre = other.genre;
    }

    // Clone method
    @Override
    public Games clone() {
        return new Games(this);
    }

    // Getters
    public String getGenre() { return genre; }

    // Setters
    public void setGenre(String genre) { this.genre = genre; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nGenre: " + genre;
    }
}

File: Task 3/C++Stock Viewer/Product/Games.cpp
#include "Games.h"
Games::Games():Product(){
    this->genre = "";
}
Games::Games(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string genre) 
            :Product(name, cost, supplier, discount_amount, packing_units, cost_per_unit){
    this->genre = genre;
}

Games::Games(const Games& other) : Product(other) {
    this->genre = other.genre;
}

std::string Games::display_product_info(){
    return Product::display_product_info()+ 
            "\nGenre: " +genre;
}